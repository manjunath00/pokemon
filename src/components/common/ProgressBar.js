import React from "react";
import PropTypes from "prop-types";

const propTypes = {
  type: PropTypes.string.isRequired,
  score: PropTypes.number.isRequired,
};

function ProgressBar({ type, score }) {
  const color = type === "Defense" ? "#1199ff" : "#ffa500";

  const style = {
    position: "absolute",
    height: "100%",
    width: `${score > 100 ? 100 : score}%`,
    borderRadius: "5px",
    animationDuration: "2s",
    animationTimingFunction: "ease-out",
    animationFillMode: "both",
    backgroundColor: color,
  };

  return (
    <div className='progressbar'>
      <div className='flex'>
        <p>{type}</p>
        <p>{score}</p>
      </div>
      <div className='progressbar__container'>
        <span style={style}></span>
      </div>
    </div>
  );
}

ProgressBar.propTypes = propTypes;

export default ProgressBar;
