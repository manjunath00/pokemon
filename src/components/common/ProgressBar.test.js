import React from "react";
import { shallow } from "enzyme";

import ProgressBar from "./ProgressBar";

it("Renders the component with two props", () => {
  const type = "Attack";
  const score = 59;
  const component = shallow(<ProgressBar type={type} score={score}/>);
  expect(component).toMatchSnapshot();
});
