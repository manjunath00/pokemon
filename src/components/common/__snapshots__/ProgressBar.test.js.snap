// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the component with two props 1`] = `ShallowWrapper {}`;
